src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -138,7 +138,9 @@ public static String escapeString(String value) {
     }
 
     public static Pattern toJavaPattern(String sqlPattern) {
-        return Pattern.compile(sqlPattern.replaceAll("%", ".*").replaceAll("_", "."));
+        return sqlPattern == null
+                ? null
```

Comment:
[nit] I'd slightly prefer creating a ".*" pattern here and then getting rid of the null check on 309 because I think `null` is equivalent to ".*" in all of the meta functions.

Reply:
@pmeredit : Creating a ".*" pattern makes things look easier, but in term of performance I think Oliver's solution is better. No need to compile the pattern and check if each table name matches if sqlPattern is null, we know that there is no need for filtering.
@bucaojit : I think we should also protect ourselves in getColumnsFromDB for the possibility of both table name and column name to be null. It looks like Tableau does not use null when calling getColums, but if someone does, we will also crash with a NPE.